pytensor/tensor/rewriting/subtensor.py

```diff
@@ -381,6 +382,58 @@ def local_useless_slice(fgraph, node):
             return [node.inputs[0]]
 
 
+@register_useless
+@register_canonicalize
+@register_stabilize
+@register_specialize
+@node_rewriter([Subtensor])
+def local_replace_slice(fgraph, node):
+    """
+    Rewrite Subtensor of the form:
+        X[0:7:1] -> X[None:None:None]
+    where X is a vector of length 7
+
+    """
+    idxs = get_idx_list(node.inputs, node.op.idx_list)
+    x = node.inputs[0]
+
+    if not idxs:
+        return
+
+    new_idxs = list(idxs)
+    idx_flag = False
+    for dim, s in enumerate(new_idxs):
+        if not isinstance(s, slice):
```

Comment:
Add a quick check to skip None slices as they should be pretty common
```suggestion
        if not isinstance(s, slice) or s == slice(None):
```